src/streaming/file.sink.cpp

```diff
@@ -3,31 +3,226 @@
 
 #include <filesystem>
 
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <cstring>
+#include <fcntl.h>
+#include <sys/uio.h>
+#include <unistd.h>
+#endif
+
 namespace fs = std::filesystem;
 
-zarr::FileSink::FileSink(std::string_view filename, bool truncate)
-  : file_(filename.data(),
-          truncate ? (std::ios::binary | std::ios::trunc) : std::ios::binary)
+namespace {
+#ifdef _WIN32
```

Comment:
If there's a way to factor out the platform-dependent things into their own source files, it makes the code much easier to read and maintain.

Reply:
100%